src/router/outputqueued/InputQueue.cc

```diff
@@ -159,19 +160,26 @@ void InputQueue::processPipeline() {
     // ensure RFE is empty
     assert(rfe_.flit == nullptr);
 
-    // pull out the front flit
+    // get the front flit
     Flit* flit = buffer_.front();
-    buffer_.pop();
 
-    // send a credit back
-    router_->sendCredit(port_, vc_);
+    // when store and forward is enabled, wait for the whole packet
+    assert(!storeAndForward_ || depth_ >= flit->packet()->numFlits());
```

Comment:
I think this would be much more readable as

if (storeAndForward_) {
assert(depth_ >= flit->packet()->numFlits());
}

Reply:
Ack